src/core/server/environment/environment_service.ts

```diff
@@ -43,17 +45,24 @@ export class EnvironmentService {
   private uuid: string = '';
 
   constructor(core: CoreContext) {
-    this.log = core.logger.get('uuid');
+    this.log = core.logger.get('environment');
     this.configService = core.configService;
   }
 
   public async setup() {
-    const [pathConfig, serverConfig] = await Promise.all([
+    const [pathConfig, serverConfig, pidConfig] = await Promise.all([
       this.configService.atPath<PathConfigType>(pathConfigDef.path).pipe(take(1)).toPromise(),
       this.configService.atPath<HttpConfigType>(httpConfigDef.path).pipe(take(1)).toPromise(),
+      this.configService.atPath<PidConfigType>(pidConfigDef.path).pipe(take(1)).toPromise(),
     ]);
 
+    // was present in the legacy `pid` file. Should be removed once we upgrade to node 16.
```

Comment:
If we down the line want to crash in production in case of an unhandled rejection, waiting to Node.js 16 isn't necessary. We could do that already now if we wanted to.

We can either choose to override this default behaviour in Node.js 16 and keep not crashing, or we can choose to crash already now or in Kibana v8.0.0. I don't see any reason to tie this decision to the upgrade to Node.js 16.